src/Arcus.Security.Providers.KeyPerFile/SecretStoreBuilderExtensions.cs

```diff
@@ -0,0 +1,19 @@
+﻿using Microsoft.Extensions.Configuration.KeyPerFile;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Hosting;
+
+namespace Arcus.Security.Providers.KeyPerFile
+{
+    public static class SecretStoreBuilderExtensions
+    {
+        public static SecretStoreBuilder AddKeyPerFile(this SecretStoreBuilder builder, string directoryPath, bool optional = true)
```

Comment:
I'll have to check with @tomkerkhove , but I think we can benefit from having also here a secret name mutation optional parameter.  See [this link](https://github.com/arcus-azure/arcus.security/blob/03a7d85b4f3b44ee1a1c2c7757202167e8357606/src/Arcus.Security.Providers.UserSecrets/Extensions/SecretStoreBuilderExtensions.cs#L30) on how this is done for the User Secrets. It makes it easier maintenance for devs. (Plus, some tests are in order when this optional parameter is added of course).

Reply:
Yes indeed. If I use "foo-bar" as a pattern, it would be good to be able to do "foo-bar" --> "foo:bar"

Reply:
The underlying Config Provider already does a translation from "foo:bar" to "foo__bar", see [this test](https://github.com/fgheysels/arcus.security/blob/frgh/feature/149_dockersecrets/src/Arcus.Security.Tests.Integration/DockerSecrets/SecretStoreBuilderExtensionTests.cs#L42), as docker secrets use the double underscore as separator.

Or are you talking about something else ?

Reply:
We provide overloads where you can provide your own translation. Maybe they are using "foo-bar" in the app and you want to look it up in the Docker Secrets through "foo:bar".